edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClient.cs

```diff
@@ -106,17 +129,19 @@ static async Task<IModuleClient> Create(Option<UpstreamProtocol> upstreamProtoco
         }
 
         static async Task<Client.ModuleClient> CreateAndOpenDeviceClient(
-            TransportType transport,
+            UpstreamProtocol upstreamProtocol,
             Option<string> connectionString,
             ConnectionStatusChangesHandler statusChangedHandler,
+            Option<IWebProxy> proxy,
             Option<string> productInfo)
         {
-            Events.AttemptingConnectionWithTransport(transport);
-            Client.ModuleClient deviceClient = await connectionString.Map(cs => Task.FromResult(Client.ModuleClient.CreateFromConnectionString(cs, transport)))
-                .GetOrElse(() => Client.ModuleClient.CreateFromEnvironmentAsync(transport));
+            ITransportSettings[] settings = { TransportMap[upstreamProtocol](proxy) };
+            Events.AttemptingConnectionWithTransport(settings[0].GetTransportType());
```

Comment:
Super nit: it might be cleaner to get settings[0] in a variable..

Reply:
Yep, will do.

Reply:
Done.